pkg/controller/deployment/deployment_helper.go

```diff
@@ -76,6 +76,17 @@ func parseArgMap(argMap map[string]string, args []string) {
 	}
 }
 
+func mergeLabels(sourceLabels map[string]string, overrideLabels map[string]string) map[string]string {
```

Comment:
Can you also add a few unit tests for this func as well (similar to the other merge funcs)?

Reply:
Even better, you can re-use https://github.com/kubernetes/apimachinery/blob/31e4b9d241f7bb8490585cc57a0e5ee5a068b432/pkg/labels/labels.go#L118

Reply:
The function signature will change everywhere to `labels.Set` but should be acceptable since there are specific functions for labels.

Reply:
Yup, updated to use `labels.Merge` function in the implementation.